src/fu-device-list.c

```diff
@@ -393,32 +380,8 @@ fu_device_list_replace (FuDeviceList *self, FuDeviceItem *item, FuDevice *device
 		item->remove_id = 0;
 	}
 
-	/* copy over any GUIDs that used to exist */
-	fu_device_list_add_missing_guids (device, item->device);
-
-	/* enforce the vendor ID if specified */
-	if (fu_device_get_vendor_id (item->device) != NULL &&
-	    fu_device_get_vendor_id (device) == NULL) {
-		const gchar *vendor_id = fu_device_get_vendor_id (item->device);
-		g_debug ("copying old vendor ID %s to new device", vendor_id);
-		fu_device_set_vendor_id (device, vendor_id);
-	}
-
-	/* copy over the version strings if not set */
-	if (fu_device_get_version (item->device) != NULL &&
-	    fu_device_get_version (device) == NULL) {
-		const gchar *version = fu_device_get_version (item->device);
-		g_debug ("copying old version %s to new device", version);
-		fu_device_set_version (device, version);
-	}
-
-	/* always use the runtime version */
-	if (fu_device_has_flag (item->device, FWUPD_DEVICE_FLAG_USE_RUNTIME_VERSION) &&
-	    fu_device_has_flag (item->device, FWUPD_DEVICE_FLAG_NEEDS_BOOTLOADER)) {
-		const gchar *version = fu_device_get_version (item->device);
-		g_debug ("forcing runtime version %s to new device", version);
-		fu_device_set_version (device, version);
-	}
+	/* copy over everything */
+	fu_device_incorporate (device, item->device);
```

Comment:
We can't do that; it's a bull in a china shop. This would break nitrokey for one, where we use two different plugins for runtime and bootloader modes. I've pushed a patch based on an older version of one of your patches, which should help -- but the takeaway I think needs to be that we need to care about setting the parent_guid on FuDevice objects that replace other things.